### ISSUE TITLE
HPA doesn't work for IstioMeshGateway objects with istio operator v2.16.0

### ISSUE THREAD

**Issue by vitalii-buchyn-exa** [banzaicloud/istio-operator#890]
**Describe the bug**
Hello, community.

We're experiencing an issue similar to https://github.com/banzaicloud/istio-operator/issues/878, but this time for IstioMeshGateway objects.

IstioMeshGateway spec.deployment.replicas example:
```
    replicas:
      count: 45
      min: 45
      max: 50
```

Deployment always stays on min count:
```
meshgateway-internal   45/45   45           45          77m
```

HPA reports:
```
meshgateway-internal   Deployment/meshgateway-internal   183%/80%   45        50        50         77m
```

Below an events example of HPA adjustments reset
```
  Normal  ScalingReplicaSet  47m (x5 over 58m)     deployment-controller  (combined from similar events): Scaled up replica set meshgateway-internal-564d677c8b to 50
  Normal  ScalingReplicaSet  39m (x3 over 47m)     deployment-controller  Scaled down replica set meshgateway-internal-564d677c8b to 45
  Normal  ScalingReplicaSet  3m41s (x13 over 43m)  deployment-controller  Scaled up replica set meshgateway-internal-564d677c8b to 50
```

Please let me know if any additional info is required.
Thank you!

**Steps to reproduce the issue:**
Deploy istio operator v2.16.0
Create IstioMeshGateway object
Load an appropriate ingress gateways
Check HPA status and an appropriate deployment replica count

**Expected behavior**
HPA for IstioMeshGateway objects to set a correct replicas count

**Additional context**
istio-operator v2.16.0
installed by helm chart version 2.1.0

**Comment by luciferinlove** [comment]
Hello @vitalii-buchyn-exa. Thank you for informing us about the problem. There was an old custom logic for autoscaling. It's not required anymore. I replaced it with the HPA default one everywhere. This should be fixed in the 2.16.1 version.

**Comment by vitalii-buchyn-exa** [comment]
@LuciferInLove 2.16.1 version works fine
thank you!
